### ISSUE TITLE
Eliminate relying on window.location for inferring userId

### ISSUE THREAD

**Issue by begedin** [coderly/ember-gdrive#33]
`app/authenticators/gdrive` is relying on window.location to retrieve the current document's id and infer the user id from that. 

Now that `ember-cli-simple-auth` is part of `ember-gdrive`, it should be possible to avoid this and implement login hint caching and retrieval in a more robust way.

**Comment by begedin** [comment]
@venkatd: Should I consider this a bit higher in priority.

If we are to use the add-on in any app, I don't think we can confidently rely on the `documentId` being parseable from window.location

I don't know if it will affect our ToDoMVC demo, but it definitely will affect the add-on's usage in actual applications.

**Comment by venkatd** [comment]
@begedin do you have any ideas on how we can handle this?

**Comment by venkatd** [comment]
My thoughts are you can get started on the TodoMVC demo and judge the importance of these tasks as you proceed with that. I think you will get a more accurate perspective when you put ember-gdrive to use. What are your thoughts?

**Comment by begedin** [comment]
I tend to agree. I wanted to make note of the possibility this might become important soon.

**Comment by venkatd** [comment]
Yeah good to have possible tasks documented so I'm all for having this written down

**Comment by begedin** [comment]
Update, now that I've worked with begedin/ember-gdriveTodoMVC a bit.

Using the location to infer the document ID requires the user using the add-.on in their app to structure their app's routes in a quite specific way.

* `locationType` in the config needs to be set to 'hash'
* the resource routes all need to be placed under a single 'document' route with it's path set to `/d/:document_id`

This is doable, but I think a bit too specific for an add-on to require. The paths at least should be freely modifiable. 

I'm not sure allowing freedom in the routing hierarchy is doable at this time.

**Comment by venkatd** [comment]
I feel that allowing them to customize the route structure isn't important now because deviating from `/d/:document_id` doesn't provide a sizable benefit. I think better to force this convention for now.

I'm open to reducing the fragility of the current solution, but if it will take a lot of work, I'd rather put it in the backlog. 

Thoughts?

**Comment by begedin** [comment]
A minor thing I could do is change the way the documetnId is parsed. Currently, it uses 
```
return location.href.split('/#/d/')[1].split('/')[0];
```

Because of that, the app is required to, in addition to having a document route with the path `/d/:document_id`, also have it's `locationType` configuration property set to 'hash'. It's an unnecessary constraint, in my opinion, and the fix would be simply to replace the above line with
```
return location.href.split('/d/')[1].split('/')[0];
```

**Comment by venkatd** [comment]
I agree. This is a good idea.
